### PR TITLE
Use VFT test in hierarchy guard converted from profiled guard with VFT test

### PR DESCRIPTION
**Pull request by liqunl**
This PR fix a problem where VP converts a profiled guard with VFT test to a hierarchy guard with method test. A method test is not sufficient as the optimizer may assume the receiver has a fixed type based on the guard and transform the code based on this assumption. One example is devirtualization.

This PR also create `ResolvedClass` constraint on a VFT load if the object class is resolved.

### CODE REVIEW

**Comment by liqunl** [comment]
@andrewcraik May I ask for your review?

File: compiler/compile/VirtualGuard.cpp [hunk 1]
```diff
@@ -133,10 +133,18 @@ TR::Node*
 TR_VirtualGuard::createVftGuard
 (TR_VirtualGuardKind kind, TR::Compilation * comp, int16_t calleeIndex,
  TR::Node* callNode, TR::TreeTop * destination, TR_OpaqueClassBlock *thisClass)
+   {
+   return createVftGuardWithReceiver(kind, comp, calleeIndex, callNode, destination, thisClass, callNode->getSecondChild());
```
Comment (andrewcraik) [inline compiler/compile/VirtualGuard.cpp]: Shouldn't this use the Node api to get the index of the child to read rather than hardcoding it?

Comment (liqunl) [reply compiler/compile/VirtualGuard.cpp]: @andrewcraik Fixed.

**Comment by andrewcraik** [comment]
@genie-omr build all

**Comment by andrewcraik** [comment]
@genie-omr build arm,win

**Pull request merged**